src/libraries/Common/src/System/Net/Security/SecurityBuffer.Windows.cs

```diff
@@ -28,6 +29,62 @@ namespace System.Net.Security
         private SecurityBuffer _item2;
     }
 
+    [StructLayout(LayoutKind.Sequential)]
+    internal ref struct InputSecurityBuffers
+    {
+        internal int Count;
+        private InputSecurityBuffer _item0;
+        private InputSecurityBuffer _item1;
+        private InputSecurityBuffer _item2;
+
+        internal InputSecurityBuffer GetBuffer(int index)
+        {
+            Debug.Assert(index < 3);
+            return index == 0 ? _item0 : index == 1 ? _item1 : _item2;
```

Comment:
It might help perf-wise if GetBuffer returned `ref InputSecurityBuffer` rather than `InputSecurityBuffer`.